app/models/usage_monitoring/alert.rb

```diff
@@ -21,7 +21,7 @@ class Alert < ApplicationRecord
     default_scope -> { kept }
 
     belongs_to :organization
-    belongs_to :billable_metric, optional: true
+    belongs_to :billable_metric, -> { with_discarded }, optional: true
```

Comment:
What do you think of loading discarded BM only for discarded alert?
```suggestion
    belongs_to :billable_metric, -> { with_discarded if discarded? }, optional: true
```